packages/capabilities/src/utils.js

```diff
@@ -57,6 +57,24 @@ export function equal(child, parent, constraint) {
   }
 }
 
+/**
+ * @param {Types.Link<unknown, number, number, 0 | 1> | undefined} child
+ * @param {Types.Link<unknown, number, number, 0 | 1> | undefined} parent
+ * @param {string} constraint
```

Comment:
would be helpful to explain this parameter. It's used to build the error message?